src/freenet/pluginmanager/PluginManager.java

```diff
@@ -876,6 +879,34 @@ public PluginInfoWrapper getPluginInfo(String plugname) {
 	}
 
 	/**
+	 * @param pluginClassName The name of the main class of the plugin - that is the class which implements {@link FredPlugin}.
+	 * @return The plugin with the given class, or null if no matching plugin was found.
+	 */
+	public PluginInfoWrapper getPluginInfoByClassName(String pluginClassName) {
+		synchronized(pluginWrappers) {
+			for(PluginInfoWrapper piw : pluginWrappers) {
+				if(piw.getPluginClassName().equals(pluginClassName))
+					return piw;
+			}
+		}
+		return null;
+	}
+
+	/**
+	 * @param pluginFileName The filename of the JAR from which the plugin was loaded.
+	 * @return Null if no matching plugin was found.
```

Comment:
Fixed by https://github.com/freenet/fred/pull/339 which had to be a separate pull request since this one is merged already.
Thanks!